lib/drivers/docker/index.js

```diff
@@ -0,0 +1,221 @@
+var sCtl = require('strong-control-channel/client');
+var Deployment = require('./deployment');
+var EventEmitter = require('events').EventEmitter;
+var fmt = require('util').format;
+var inherits = require('util').inherits;
+var sdr = require('strong-docker-run');
+var _ = require('lodash');
+
+module.exports = exports = DockerDriver;
+
+function DockerDriver(opts) {
+  if (!opts.baseDir || !opts.console || !opts.server) {
+    throw Error('Requires baseDir, console, and server');
+  }
+  if (!(this instanceof DockerDriver)) {
+    return new DockerDriver(opts);
+  }
+  EventEmitter.call(this);
+
+  // XXX(rmg) can I put temp files here?
+  this.baseDir = opts.baseDir;
+  // XXX(rmg) for log messages? or something else?
```

Comment:
Existing code has some amount of verbosity in non-debug mode (deploys, errors, major actions), so even without `DEBUG`, you can see the log and have an idea what's going on. But direct calls to console go directly to console... hard to trap in unit tests, so I started passing the console object in as an argument to contructors of objects that are going to use console, so that it can be injected if needed. Technically, pm could switch to a log framework, and inject that framework in as the console option (assuming it was api compatible), and we wouldn't have to do anything to the  driver to support that switch. That's my thinking, anyway.